backend/wmg/api/v2.py

```diff
@@ -101,9 +101,25 @@ def query():
                 expression_summary, cell_counts, group_by_terms
             )
             if is_rollup:
-                rolled_gene_expression_df, rolled_cell_counts_grouped_df = rollup(
-                    gene_expression_df, cell_counts_grouped_df
+                # do not filter out redundant nodes for gene expressions. certain cell types may only
+                # appear redundant because they do not express a particular gene and are thus missing
+                # from the gene expression dataframe.
+                rolled_gene_expression_df = rollup(
+                    gene_expression_df, snapshot.cell_type_ancestors, filter_redundant_nodes=False
                 )
+                rolled_cell_counts_grouped_df = rollup(cell_counts_grouped_df, snapshot.cell_type_ancestors)
+
+                # filter out rows that do not have a corresponding cell type in the cell counts dataframe
+                # as these will not be displayed in the UI anyway
+
+                with ServerTiming.time("filter out rows"):
```

Comment:
This block filters out redundant cell types from gene expression by checking to see what's not included in the cell counts dataframe, since the cell counts rollup DOES filter out redundant cell types (see above comment for more info).